tf/edgeml/trainer/bonsaiTrainer.py

```diff
@@ -389,18 +432,61 @@ def train(self, batchSize, totalEpochs, sess,
 
             # This helps in direct testing instead of extracting the model out
 
-            testAcc, testLoss, regTestLoss = sess.run(
-                [self.accuracy, self.loss, self.regLoss], feed_dict=_feed_dict)
+            testAcc, testLoss, regTestLoss, pred = sess.run(
+                [self.accuracy, self.loss, self.regLoss, self.prediction], feed_dict=_feed_dict)
+
             if ihtDone == 0:
-                maxTestAcc = -10000
-                maxTestAccEpoch = i
-            else:
-                if maxTestAcc <= testAcc:
+                if (self.bonsaiObj.isRegression == False):
+                    maxTestAcc = -10000
                     maxTestAccEpoch = i
+                elif (self.bonsaiObj.isRegression == True):
                     maxTestAcc = testAcc
-                    self.saveParams(currDir)
+                    maxTestAccEpoch = i
+
+            else:
+                if (self.bonsaiObj.isRegression == False):
+                    if maxTestAcc <= testAcc:
+                        maxTestAccEpoch = i
+                        maxTestAcc = testAcc
+                        self.saveParams(currDir)
+                elif (self.bonsaiObj.isRegression == True):
+                    print("Minimum Training MAE : ", np.mean(maxTestAcc))
+                    if maxTestAcc >= testAcc:
+                        # For regression , we're more interested in the minimum MAE.
+                        maxTestAccEpoch = i
+                        maxTestAcc = testAcc
+                        self.saveParams(currDir)
+            if(i == totalEpochs-1):
+                if (self.bonsaiObj.isRegression == True):
+                    print("TESTING")
+                    print("-------------------------------------------")
+                    pred = pred[0]
+                    print(np.transpose(pred).shape, Ytest.shape)
```

Comment:
Please remove these testing print comments. You don't need to output the shapes.

Reply:
Removed printing the shapes.